dask/dataframe/groupby.py

```diff
@@ -1023,6 +1023,15 @@ def _cumcount_aggregate(a, b, fill_value=None):
     return a.add(b, fill_value=fill_value) + 1
 
 
+def _fillna_groups(groups, axis, **kwargs):
+    if axis == 0:
+        return groups.fillna(axis=axis, **kwargs)
+    else:
+        return groups.drop(columns=groups.columns[0], axis=1).fillna(
```

Comment:
@ian-r-rose Is this way of dropping columns alright? I noticed there is a `drop_columns` dataframe-method, would it make sense to use that instead?

https://github.com/dask/dask/blob/1b1a96b4ddf31d0879e718be463a731af7f1d086/dask/dataframe/methods.py#L379-L382

Reply:
And, I'm also not sure about bundling drop-column logic in the top-level `def _fillna_groups()`. is this OK, or would it be better to do this in `fillna()` (line 2014)?

Reply:
Just to note: I'm yet to add doctrings and comments :)

Reply:
I think using `df.drop` directly is okay. It looks like the `drop_columns` utility is primarily used for `map_partitions`, which doesn't really apply here.

However, I think always dropping the zeroth column doesn't quite do what we want:
1. The column we want to drop might not be the first one.
2. The user might be grouping by multiple columns

What if instead we also included a `by` argument in `_fillna_group` for the column(s) by which we are grouping, the dataframe can pass in `self.by`, and then we can drop them?

Reply:
> What if instead we also included a by argument in _fillna_group for the column(s) by which we are grouping, the dataframe can pass in self.by, and then we can drop them?

Yes! Thanks for this suggestion!